pkg/container/docker.go

```diff
@@ -184,7 +199,20 @@ func (f *BuilderFactory) GenerateImageHash() (string, error) {
 		return "", ErrHashingBuildContext.Wrap(err)
 	}
 
-	logrus.Debug("Generated image hash: ", fmt.Sprintf("%x", hasher.Sum(nil)))
+	f.logger.Debug("Generated image hash: ", fmt.Sprintf("%x", hasher.Sum(nil)))
 
 	return fmt.Sprintf("%x", hasher.Sum(nil)), nil
 }
+
+func verifyOptions(opts BuilderFactoryOptions) error {
+	if opts.ImageName == "" {
+		return ErrImageNameEmpty
+	}
+	if opts.BuildContext == "" {
+		return ErrBuildContextEmpty
+	}
+	if opts.Logger == nil {
```

Comment:
Same here, if a logger isn't passed in, can't we just create one?

Reply:
The logger is required throughout the project so we use a unique logger object to handle all the logs the same way. i.e. it is more consistent to have only one logger object and pass it throughout the entire code base.

Reply:
fwiw making it required here isn't the same as ensuring it is the same across the code base. 
but I won't block on this. You can resolve this.

Reply:
an addition point is the fact that this BuilderFactory is used internally by the knuu and the external user is not exposed to it. So, we have more flexibility to refactor it the way we think is optimized in future.